usr/share/rear/conf/default.conf

```diff
@@ -4018,14 +4018,47 @@ UEFI_BOOTLOADER=""
 # then ReaR will try to guess the shim.efi and set it as SECURE_BOOT_BOOTLOADER.
 SECURE_BOOT_BOOTLOADER=""
 #
-# Some backup tools cannot store vfat filesystems and therefore also cannot
-# restore the EFI System Partition (ESP) that must be a vfat filesystem.
-# For those, ReaR can backup and restore the ESP via the rescue image.
-# However, this is not recommended to do by default as the rescue image
-# could be older than the latest backup and the restored ESP could be
-# outdated or incompatible with the restored system. We enable this automatically
-# only for problematic backup software that cannot store vfat filesystems.
+# RESTORE_ESP_FROM_RESCUE_SYSTEM
+#
+# Some backup tools cannot backup VFAT filesystems and therefore also
+# cannot restore files on the EFI System Partition (ESP) which is a VFAT filesystem.
+# For those, ReaR can backup and restore the ESP via the ReaR rescue/recovery system.
+# It is not recommended to do this by default as the ReaR rescue/recovery system
+# could be older than the latest backup with the backup tool
+# so a restored ESP from the ReaR rescue/recovery system could be outdated
+# or incompatible with a restored backup which could make UEFI booting fail.
+# We enable this automatically only for known problematic backup software
+# that cannot backup and restore VFAT filesystems (currently only for BACKUP=PPDM).
 RESTORE_ESP_FROM_RESCUE_SYSTEM=no
+#
+# EFIBOOTMGR_CREATE_ENTRIES
+#
+# EFIBOOTMGR_CREATE_ENTRIES is an array of strings which specify the UEFI Boot Manager entries
+# that need to be created by 'efibootmgr' at the end of "rear recover" on the recreated system
+# to make the recreated system UEFI bootable. For example
+# EFIBOOTMGR_CREATE_ENTRIES=( '/dev/sda 1 EFI\boot\shim.efi shim on sda1' '/dev/sdb 2 EFI\boot\bootx64.efi bootx64 on sdb2' )
+# would let 'efibootmgr' create two UEFI Boot Manager entries to launch two EFI binaries:
+# One entry labeled 'shim on sda1' for an EFI binary shim.efi which is located as file 'EFI\boot\shim.efi'
+# in the VFAT filesystem of the EFI System Partition (ESP) which is partition number '1' on the disk /dev/sda
+# and another one labeled 'bootx64 on sdb2' for the EFI binary 'EFI\boot\bootx64.efi' on another ESP /dev/sdb2.
+# The syntax of the strings which specify the UEFI Boot Manager entries is as follows:
+# The first word specifies the disk device node on which the ESP is.
+# The second word is the partition number of the ESP.
+# The third word is the EFI binary file name in the VFAT filesystem on the ESP.
+# All remaining words are the label of the UEFI Boot Manager entry.
+# At least the disk device node on which the ESP is must be specified.
+# When the partition number of the ESP is not specified or when it is specified as '0'
+# then '1' will be used as partition number by default.
```

Comment:
I think that this is also a suboptimal interface. If the user wants `1`, let them specify `1`. I would reserve "no value" or a special value like `0` for autodetection. I.e. When the partition number of the ESP is not specified, then the partition number of the ESP on the disk will be used.

Reply:
@pcahyna 
can you provide improved code that does what you ask for?

My reasoning for my current behaviour:
Currently there is no autodetection for EFIBOOTMGR_CREATE_ENTRIES
(@pcahyna fell free to add it later as you like).
If autodetection would be implemented for EFIBOOTMGR_CREATE_ENTRIES
then the value `0` should be used for autodetection.
Because currently there is no autodetection, the value `0`
is currently used for a hardcoded default value.
The difference is:
When the suer specifes `1` his specified value is used
(ReaR has to obey what his user explicitly specified).
When the user specifes `0` he gets a value from the system
either some default value or - when implemented - some
autodetection value.